libvirt/resource_libvirt_domain.go

```diff
@@ -528,6 +535,76 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	// network-installation used in combination with kernel and initrd parameters.
+	NetworkAutoinstallEnabled := d.Get("network_autoinstall").(bool)
+	if NetworkAutoinstallEnabled {
+		stop := make(chan int)
+		rebootCallBack := func(c *libvirt.Connect, d *libvirt.Domain) {
+			log.Printf("[DEBUG:] Libvirt-events: Caught reboot event!")
+			err := destroyDomain(domain)
+			if err != nil {
+				panic(err)
+			}
+			domainDef, _ := getXMLDomainDefFromLibvirt(domain)
+			domainDef.OS.Kernel = ""
```

Comment:
This completely changes the semantics of _initrd_ and _kernel_, as they will be gone after reboot, and then terraform would attempt to re-add them.

Reply:
Yes, but there is no workaround for it. We need to remove them

Reply:
I think the user should be aware that he need.to change the tf file